airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/datetime_stream_slicer.py

```diff
@@ -173,15 +174,13 @@ def _get_date(self, cursor_value, default_date: datetime.datetime, comparator) -
         cursor_date = self.parse_date(cursor_value or default_date)
         return comparator(cursor_date, default_date)
 
-    def parse_date(self, date: Any) -> datetime:
-        if date and isinstance(date, str):
-            if self.is_int(date):
-                return datetime.datetime.fromtimestamp(int(date)).replace(tzinfo=self._timezone)
-            else:
-                return dateutil.parser.parse(date).replace(tzinfo=self._timezone)
-        elif isinstance(date, int):
+    def parse_date(self, date: Union[str, datetime.datetime]) -> datetime.datetime:
+        if self._datetime_format == TIMESTAMP_FORMAT:
             return datetime.datetime.fromtimestamp(int(date)).replace(tzinfo=self._timezone)
-        return date
+        elif isinstance(date, str):
+            return datetime.datetime.strptime(str(date), self._datetime_format).replace(tzinfo=self._timezone)
```

Comment:
better to use the specified datetime_format instead of relying on dateutil.parser's best guess